engine/collection/compliance/engine.go

```diff
@@ -90,53 +89,6 @@ func NewEngine(
 	return eng, nil
 }
 
```

Comment:
wondering if we still need the field `Engine.cm`: https://github.com/onflow/flow-go/blob/6a6b5eb8f5361813623ff4a26944590fb9b45716/engine/collection/compliance/engine.go#L43

I think the only place where we use it is in the constructor, where we assign its value to `eng.Component`: https://github.com/onflow/flow-go/blob/6a6b5eb8f5361813623ff4a26944590fb9b45716/engine/collection/compliance/engine.go#L82-L87

### suggestion:
could apply the same revision here, as you did for the [main consensus compliance engine](https://github.com/onflow/flow-go/blob/6a6b5eb8f5361813623ff4a26944590fb9b45716/engine/consensus/compliance/engine.go#L47)
<img width="1077" alt="Screen Shot 2022-11-21 at 8 32 56 PM" src="https://user-images.githubusercontent.com/26322303/203222234-310c5bff-8740-423d-90a0-bd7ba99caf19.png">